lightning-graph-sync/src/lib.rs

```diff
@@ -0,0 +1,259 @@
+#![deny(missing_docs)]
+#![deny(unsafe_code)]
+#![deny(broken_intra_doc_links)]
+#![deny(non_upper_case_globals)]
+#![deny(non_camel_case_types)]
+#![deny(non_snake_case)]
+#![deny(unused_mut)]
+#![deny(unused_variables)]
+#![deny(unused_imports)]
+
+use std::fs::File;
+
+use lightning::routing::network_graph;
+use lightning_block_sync::http::*;
+
+use crate::error::GraphSyncError;
+
+pub mod error;
+pub mod processing;
+
+pub async fn sync_network_graph_with_url(
+	network_graph: &network_graph::NetworkGraph,
+	sync_host: &str,
+	sync_port: u16,
+	sync_path: &str,
+) -> Result<(), GraphSyncError> {
+	// make sure there is precisely one leading slash
+	let canonical_path = format!("/{}", sync_path.trim_start_matches("/"));
+	let http_endpoint = HttpEndpoint::for_host(sync_host.into()).with_port(sync_port);
+	let http_client_result = HttpClient::connect(&http_endpoint);
+	let mut http_client = http_client_result?;
+
+	let response_result = http_client
+		.get::<BinaryResponse>(&canonical_path, sync_host)
+		.await;
+
+	let response = response_result?;
+	let response_bytes = response.0;
+
+	let update_result = processing::update_network_graph(&network_graph, &response_bytes[..])?;
+	Ok(update_result)
```

Comment:
It's a bit more idiomatic to forgo the intermediary variables:

```rust
let mut http_client = HttpClient::connect(&http_endpoint)?;
let response_bytes = http_client.get::<BinaryResponse>(&canonical_path, sync_host).await?.0;
processing::update_network_graph(&network_graph, &response_bytes[..], &chain_source)
```

Reply:
sounds good

Reply:
done